client/src/api/decksAPI.js

```diff
@@ -1,10 +1,15 @@
 import apiRequest from "./index.js";
 
-export const getDecks = async ({ page = 1, limit = 10 } = {}) => {
+export const getDecks = async ({ page = 1, limit = 10, search = "" } = {}) => {
   const params = new URLSearchParams({
     page: String(page),
     limit: String(limit),
   });
+
+  if (search) {
+    params.set("search", search);
+  }
```

Comment:
Since you already added a default value for `search`, the `if` statement will always be true. 
Therefore, simply add the `search` to the params directly


```suggestion
    search: String(search),
});
```